crates/starknet-devnet/tests/test_get_class.rs

```diff
@@ -279,4 +282,69 @@ mod get_class_tests {
             }
         }
     }
+
+    #[tokio::test]
+    async fn test_getting_compiled_casm_for_cairo0_or_non_existing_hash_have_to_return_class_hash_not_found_error()
+     {
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&["--account-class", "cairo0"])
+            .await
+            .expect("Could not start Devnet");
+
+        let (_, account_address) = devnet.get_first_predeployed_account().await;
+
+        let block_id = BlockId::Tag(BlockTag::Latest);
+
+        let class_hash =
+            devnet.json_rpc_client.get_class_hash_at(block_id, account_address).await.unwrap();
+
+        // Felt::ONE is non existing class hash
+        for el in [class_hash, Felt::ONE] {
+            match get_compiled_casm(&devnet, block_id, el).await.unwrap_err() {
+                StarknetError::ClassHashNotFound => {}
+                other => panic!("Unexpected error {:?}", other),
+            }
+        }
+    }
+
+    #[tokio::test]
+    async fn test_getting_compiled_casm_for_cairo_1_have_to_succeed() {
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&["--account-class", "cairo1"])
+            .await
+            .expect("Could not start Devnet");
+
+        let (_, account_address) = devnet.get_first_predeployed_account().await;
+
+        let block_id = BlockId::Tag(BlockTag::Latest);
+
+        let class_hash =
+            devnet.json_rpc_client.get_class_hash_at(block_id, account_address).await.unwrap();
+
+        get_compiled_casm(&devnet, block_id, class_hash).await.unwrap();
```

Comment:
Why not assert the returned value? E.g. the request could be succeeding if we mistakenly return the `Default` implementation of `CasmContractClass` from the endpoint.

Reply:
[1a0202b](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/633/commits/1a0202b69d4e9e4a4e27783daf907164b7870b7c)

Reply:
[f885309](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/633/commits/f8853098440f38abdc7f7d88b23e947b7e9fe659)